stepflow-rs/crates/stepflow-cli/tests/commands/snapshots/test_cli__commands__test_infer__infer_secret_propagation.snap

```diff
@@ -0,0 +1,65 @@
+---
+source: crates/stepflow-cli/tests/commands/test_infer.rs
+info:
+  program: stepflow
+  args:
+    - "--log-file=/dev/null"
+    - "--omit-stack-trace"
+    - infer
+    - "--flow=stepflow-rs/crates/stepflow-cli/tests/commands/secret_propagation.yaml"
+    - "--show-step-inputs"
+  env:
+    STEPFLOW_LOG_DESTINATION: ""
+    STEPFLOW_LOG_FILE: ""
+    STEPFLOW_LOG_FORMAT: ""
+    STEPFLOW_LOG_LEVEL: ""
+    STEPFLOW_OTHER_LOG_LEVEL: ""
+    STEPFLOW_OTLP_ENDPOINT: ""
+    STEPFLOW_TRACE_ENABLED: ""
+---
+success: false
+exit_code: 1
+----- stdout -----
+📊 Validation results: 0 fatal, 2 errors, 0 warnings
+
+❌ Errors:
+  ERROR Undefined variable 'api_key' referenced in flow ()
+  ERROR Undefined variable 'temperature' referenced in flow ()
+schema: https://stepflow.org/schemas/v1/flow.json
+name: Secret Propagation Test
+description: Test that is_secret annotation propagates through variable references
+schemas:
+  type: object
+  properties:
+    output:
+      type: object
+      properties:
+        messages: {}
+      required:
+      - messages
+steps:
+- id: create_message
+  component: /builtin/create_messages
+  input:
+    user_prompt: Using API key
+    metadata:
+      api_key:
+        $variable: $.api_key
+      temperature:
+        $variable: $.temperature
+output:
+  messages:
+    $step: create_message
+    path: $.messages
+
+
+----- stderr -----
+
+📥 Synthesized step input types:
+  create_message: { user_prompt: string, metadata: { api_key: any, temperature: any } }
+
+📋 Component expected input types:
+  create_message: any
```

Comment:
Should fix the display of inferred types here, and also indicate whether things are secrets?